core/src/main/scala/chisel3/RawModule.scala

```diff
@@ -211,10 +211,11 @@ abstract class RawModule extends BaseModule {
       case (true, false) if left.probeInfo.get.writable => ProbeDefine(si, left.lref, RWProbeExpr(Node(right)))
       case (true, false)                                => ProbeDefine(si, left.lref, ProbeExpr(Node(right)))
       case (false, true)                                => Connect(si, left.lref, ProbeRead(Node(right)))
-      case (false, false) =>
+      case (false, false)                               =>
+        // For non-probe, directly create Nodes for lhs, skipping visibility check to support BoringUtils.drive.
         (left, right) match {
-          case (_: Property[_], _: Property[_]) => PropAssign(si, left.lref, Node(right))
-          case (_, _) => Connect(si, left.lref, Node(right))
+          case (_: Property[_], _: Property[_]) => PropAssign(si, Node(left), Node(right))
+          case (_, _) => Connect(si, Node(left), Node(right))
```

Comment:
I don't think node-of-non-passive (e.g., property or probe) are valid and regardless they don't currently work in CIRCT.

I'm not sure what to suggest here -- probably a wire? -- and if this path isn't already tested can you add one for it?

Reply:
There is a difference between Chisel internals Node and `node` appearing in the IR. The `lref` method on Data produces a Chisel internals Node already, this is just skipping over some visibility checking that `lref` does, which is expected to fail in the context BoringUtils.drive uses this method. Both of the happy path tests I added go through this code path, and but no `node`s are emitted into the FIRRTL.

Reply:
Awesome! Thanks, appreciate the explanation and glad to hear it!